packages/contentstack/src/hooks/prerun/latest-version-warning.ts

```diff
@@ -0,0 +1,60 @@
+import { cliux, configHandler, HttpClient, LoggerService } from '@contentstack/cli-utilities';
+import * as semver from 'semver';
+interface ICacheData {
+  lastChecked: number;
+  lastWarnedDate: string;
+  latestVersion: string;
+}
+interface IVersionUpgradeWarningFrequency {
+  versionSyncDuration: number;
+}
+const versionUpgradeWarningFrequency: IVersionUpgradeWarningFrequency = {
+  versionSyncDuration: 3 * 24 * 60 * 60 * 1000,
+};
+export default async function (_opts): Promise<void> {
+  const now = Date.now();
+  const today = new Date().toISOString().split('T')[0];
+  let logger!: LoggerService;
+  logger = new LoggerService(process.cwd(), 'cli-log');
+  let cache: ICacheData = { lastChecked: 0, lastWarnedDate: '', latestVersion: '' };
+
+  if (!configHandler.get('versionUpgradeWarningFrequency')) {
+    configHandler.set('versionUpgradeWarningFrequency', versionUpgradeWarningFrequency);
+  }
+  const versionUpgradeWarningFrequencyConfig: IVersionUpgradeWarningFrequency = configHandler.get(
+    'versionUpgradeWarningFrequency',
+  );
+
+  // Load cache if it exists
+  if (configHandler.get('versionUpgradeWarningCache')) {
+    cache = configHandler.get('versionUpgradeWarningCache');
+  }
+
+  // Perform update check if needed
+  const httpClient = new HttpClient();
+
+  if (now - cache.lastChecked > versionUpgradeWarningFrequencyConfig.versionSyncDuration) {
+    try {
+      const latestVersion = (await httpClient.get(`https://registry.npmjs.org/@contentstack/cli/latest`)).data.version;
+      cache.latestVersion = latestVersion;
+      cache.lastChecked = now;
+
+      // Save updated cache
+      configHandler.set('versionUpgradeWarningCache', cache);
+    } catch (error) {
+      logger.error('Failed to check the latest version', error);
```

Comment:
@naman-contentstack , are we displaying error on console as well? If so, it is not a good idea to throw the entire api error on the console.

Reply:
No we are not throwing the error on console, only in the log file

Reply:
logger will display in the console ?

Reply:
No, logger will display in the log file. specifically `cli-log.log` file